.github/workflows/codeql.yml

```diff
@@ -0,0 +1,57 @@
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ "7.0", "3.13", "3.14", "3.15", "3.16" ]
+  pull_request:
+    branches: [ "7.0" ]
+  schedule:
+    - cron: "2 10 * * 0"
+
+jobs:
+  analyze:
+    name: Analyze
+    runs-on: ubuntu-latest
+    permissions:
+      actions: read
+      contents: read
+      security-events: write
+
+    strategy:
+      fail-fast: false
+      matrix:
+        language: [ python, cpp ]
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Install Packages (cpp)
+        if: ${{ matrix.language == 'cpp' }}
+        run: |
+          sudo apt-get update
+          sudo apt-get install --yes libreadline-dev
+
+      - name: Initialize CodeQL
+        uses: github/codeql-action/init@v2
+        with:
+          languages: ${{ matrix.language }}
+          config-file: ./.github/codeql.yml
```

Comment:
Need to find out what the config-file actually configures. The contents of the "path_classifiers" section of our `.lgtm.yml` file isn't represented in this PR at all yet.

Reply:
Removed that file and config for now, although we might need to add it back again to disable or configure the checks it does.